packages/rollup/src/plugins/with-nx/with-nx.ts

```diff
@@ -216,19 +227,32 @@ export function withNx(
       }),
       image(),
       json(),
-      // Needed to generate type definitions, even if we're using babel or swc.
-      require('rollup-plugin-typescript2')({
-        check: !options.skipTypeCheck,
-        tsconfig: tsConfigPath,
-        tsconfigOverride: {
-          compilerOptions: createTsCompilerOptions(
-            projectRoot,
-            tsConfig,
-            options,
-            dependencies
-          ),
-        },
-      }),
+      // TypeScript compilation and declaration generation
+      // TODO(Colum): Change default value of useLegacyTypescriptPlugin to false for Nx 23
```

Comment:
I think we can be more aggressive. Change it in v22, and then remove in v23. The plugin is unmaintained and unofficial.

Can we also use `TODO(v22)` and `TODO(v23)` so it's easier to find all of the TODOs?